docs/guides/development/nodejs/nodejs-twitter-bot/index.md

```diff
@@ -0,0 +1,294 @@
+</br>
+
+
+Now, we will create some variables for these keys you’ll be copying from the Twitter developer page later. You’ll have four keys you need in order to connect your app to the code you’ve written. Create a file in your project called `.env`. Inside, add four variables: `access_token`, `access_token_secret`, `consumer_secret`, `consumer_key`. The consumer keys will pair with the API keys.
+
+Something important about a dotenv file, you won’t need quotes around the values of your variables, as long as there aren’t spaces. Make sure to write it like this: `consumer_key=whateveryourkeyis`. Your dotenv file is going to be your one source of truth for the keys. The `config.js` file will point to the `.env` values, and because you created a `gitignore`, you won’t upload your keys to GitHub. Head back to the twitter developer page and click the keys, and copy and paste them into the appropriate variables in your `.env` file.
+
+Up next, we’re going to install all the npm packages we need, so enter the following into your terminal in VSCode.
+
+```
+"Npm init -y"
+
+"Npm install node"
+
+"Npm install node-schedule"
+
+"Npm install twit"
+
+"Npm install dotenv"
+```
+</br>
+This installs all the dependencies we’ll need to be able to have a functional Twitter bot that can read the API, post on a schedule, use the `.env` file, and use `node.js`. We have one more thing to adjust, and that’s to create some scripts inside the `package.json` file code block. Open it and add the following to the scripts section.
+
+```javascript
+	"test": "index.js",
+	      "start": "node ./index.js",
+	      "develop": "NODE_ENV=develop node index.js"
+```
+</br>
+That `index.js` file doesn’t exist yet, so go ahead and create it and name it whatever you like. I’ll be calling mine `snes.js`. Go ahead and make a `config.js` at the same time and input the following into it.
+
+```javascript
+	module.exports = {
+	    consumer_key: process.env.consumer_key,
+	    consumer_secret: process.env.consumer_secret,
+	    access_token: process.env.access_token,
+	    access_token_secret: process.env.access_token_secret,
+	};
+```
+
+After all this, open up `index.js`, or whatever you've named it, and get ready to type some javascript
+
+## Code It Up.
+
+```javascript
+	console.log("SNES Soundtracks booting up");
+
+	//making sure npm run develop works
+	if (process.env.NODE_ENV === "develop") {
+	    require("dotenv").config();
+	};
+
+	//rules for node-schedule
+	var schedule = require("node-schedule");
+	var rule = new schedule.RecurrenceRule();
+	  rule.dayOfWeek = 1,
+	  rule.hour = 10;
+	  rule.tz = "Etc/GMT+4";
+
+	//array to pull soundtracks from
+	var soundtrackArray = [ "an array of youtube video URLs"];
+	var soundtrackArrayLength = soundtrackArray.length;
+	var soundtrackArrayElement = Math.floor(Math.random() * soundtrackArrayLength);
```

Comment:
I ran into an issue with this line:

By picking out the `soundtrackArrayElement` at the start of the code, it would remain the same every time pressStart or pressSelect was called, so every tweet would reference the same SNES soundtrack link.

To fix this, I removed this line and the copied it into the `pressSelect` and `pressStart` functions, just before those functions compose the tweet text

Reply:
I discovered this on my own a few weeks ago and forgot to change the submission to y'all. Great catch.